packages/sanity/src/desk/panes/document/DocumentPaneProvider.tsx

```diff
@@ -616,6 +646,17 @@ export const DocumentPaneProvider = memo((props: DocumentPaneProviderProps) => {
     }
   }, [connectionState, pushToast])
 
+  /**
+   * If the document was deleted successfully, close the pane.
+   */
+  useEffect(() => {
+    if (!documentOperationEvent) return
+    if (documentOperationEvent.type === 'success' && documentOperationEvent.op === 'delete') {
+      // Wait until next tick to allow deletion toasts to appear first
+      setTimeout(() => paneRouter.closeCurrentAndAfter(), 0)
```

Comment:
even though it's unlikely to be used it would be good if we could cancel the timeout in effect cleanup (I'm wary of subtle react strict mode issues)